### PR TITLE
[PHP] Add support for async requests

### LINKED ISSUE #5306: [PHP] Add support for async requests
##### Description
Generated clients are already using guzzle6 out of the box on 2.3.0 branch and guzzle supports async requests. It is easy task now to allow generated client to send async requests. I would suggest adding separate `$api->xxxAsync()` and `$api->xxxAsyncWithHttpInfo()` methods for this. 

##### Swagger-codegen version
2.3.0+ 
##### Command line used for generation
`--library=none` or something
##### Related issues
https://github.com/swagger-api/swagger-codegen/pull/5190
https://github.com/swagger-api/swagger-codegen/issues/1482
##### Suggest a Fix

### PR DESCRIPTION
**Pull request by ackintosh**
closes #5306 

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

(details of the change, additional tests that have been done, reference to the issue for tracking, etc)

#### Async requests

- `xxxAsync`
- `xxxAsyncWithHttpInfo`

#### Usage

```php
$promise = $api->getPeByIdtAsyncWithHttpInfo(1)->then(function ($response) {
    var_dump($response);
    // array(3) {
    //   [0] =>
    //   class Swagger\Client\Model\Pet
    //   [1] =>
    //   int(200)
    //   [2] =>
    //   array(7) {
    //     'Date' =>
    // ...
}, function ($exception) {
    var_dump($exception);
    // class Swagger\Client\ApiException
});

var_dump($promise);
// class GuzzleHttp\Promise\Promise

$promise->wait();
```

### CODE REVIEW

**Comment by wing328** [comment]
@ackintosh thanks for the enhancement.

cc @baartosz @dkarlovi @arnested

File: samples/client/petstore-security-test/php/SwaggerClient-php/lib/Api/FakeApi.php [hunk 1]
```diff
@@ -108,14 +108,101 @@ public function testCodeInjectEndRnNR($test_code_inject____end____rn_n_r = null)
...
+    }
+
+    /**
+     * Operation testCodeInjectEndRnNRAsync
+     *
+     * To test code injection *_/ ' \" =end -- \\r\\n \\n \\r
+     *
+     * @param string $test_code_inject____end____rn_n_r To test code injection *_/ &#39; \&quot; &#x3D;end -- \\r\\n \\n \\r (optional)
+     * @throws \InvalidArgumentException
+     * @return void
```
Comment (baartosz) [inline samples/client/petstore-security-test/php/SwaggerClient-php/lib/Api/FakeApi.php]: PromiseInterface is returned here, annotation is wrong

File: samples/client/petstore-security-test/php/SwaggerClient-php/lib/Api/FakeApi.php [hunk 2]
```diff
@@ -108,14 +108,101 @@ public function testCodeInjectEndRnNR($test_code_inject____end____rn_n_r = null)
...
+    }
+
+    /**
+     * Operation testCodeInjectEndRnNRAsyncWithHttpInfo
+     *
+     * To test code injection *_/ ' \" =end -- \\r\\n \\n \\r
+     *
+     * @param string $test_code_inject____end____rn_n_r To test code injection *_/ &#39; \&quot; &#x3D;end -- \\r\\n \\n \\r (optional)
+     * @throws \InvalidArgumentException
+     * @return \GuzzleHttp\Promise\Promise
```
Comment (baartosz) [inline samples/client/petstore-security-test/php/SwaggerClient-php/lib/Api/FakeApi.php]: it returns \GuzzleHttp\Promise\PromiseInterface

**Comment by baartosz** [comment]
@wing328 tag 2.2.3 is probably wrong as this branch depends on guzzle which is in 2.3.0 only

**Comment by wing328** [comment]
@baartosz already labelled as 2.3.0. Can you check again?

**Comment by baartosz** [comment]
@wing328  Sorry, I though about milestone:
`@wing328 wing328 modified the milestone: v2.3.0, v2.2.3 2 days ago`
it looks like mistake for me.

**Comment by wing328** [comment]
@baartosz no worry and thanks for reviewing the change.

**Comment by ackintosh** [comment]
@baartosz  Thanks for your comment :sparkles: I fixed it.

**Comment by baartosz** [comment]
I wrote some tests for this feature yesterday, please add them to this PR. Or I'll create another PR later.

`samples/client/petstore/php/SwaggerClient-php/tests/AsyncTest.php`:
```
<?php

namespace Swagger\Client;

use Swagger\Client\Api\PetApi;
use Swagger\Client\Model\Pet;

class AsyncTest extends \PHPUnit_Framework_TestCase
{
    /** @var PetApi */
    private $api;

    /** @var  int */
    private $petId;

    public function setUp()
    {
        $this->api = new Api\PetApi();

        $this->petId = 10005;
        $pet = new Model\Pet;
        $pet->setId($this->petId);
        $pet->setName("PHP Unit Test");
        $pet->setPhotoUrls(array("http://test_php_unit_test.com"));
        // new tag
        $tag= new Model\Tag;
        $tag->setId($this->petId); // use the same id as pet
        $tag->setName("test php tag");
        // new category
        $category = new Model\Category;
        $category->setId($this->petId); // use the same id as pet
        $category->setName("test php category");

        $pet->setTags(array($tag));
        $pet->setCategory($category);

        $pet_api = new Api\PetApi();
        // add a new pet (model)
        $add_response = $pet_api->addPet($pet);
    }

    public function testAsyncRequest()
    {
        $promise = $this->api->getPetByIdAsync(10005);

        $promise2 = $this->api->getPetByIdAsync(10005);

        $pet = $promise->wait();
        $pet2 = $promise2->wait();
        $this->assertInstanceOf(Pet::class, $pet);
        $this->assertInstanceOf(Pet::class, $pet2);
    }

    public function testAsyncRequestWithHttpInfo()
    {
        $promise = $this->api->getPetByIdAsyncWithHttpInfo($this->petId);

        list($pet, $status, $headers) = $promise->wait();
        $this->assertEquals(200, $status);
        $this->assertInternalType('array', $headers);
        $this->assertInstanceOf(Pet::class, $pet);
    }

    public function testAsyncThrowingException()
    {
        $this->setExpectedException(ApiException::class);

        $promise = $this->api->getPetByIdAsync(0);
        $promise->wait();
    }

    public function testAsyncApiExceptionWithoutWaitIsNotThrown()
    {
        $promise = $this->api->getPetByIdAsync(0);
        sleep(1);
    }

    public function testAsyncHttpInfoThrowingException()
    {
        $this->setExpectedException(ApiException::class);

        $promise = $this->api->getPetByIdAsyncWithHttpInfo(0);
        $promise->wait();
    }
}
```

**Comment by wing328** [comment]
@ackintosh thanks for the PR, which has been merged into master.

@baartosz thanks for reviewing the change and adding the test cases to cover the change.

**Pull request merged**